kyo-prelude/shared/src/main/scala/kyo/kernel/Pending.scala

```diff
@@ -162,7 +162,11 @@ object `<`:
     // TODO Compiler crash if inlined
     extension [A, S](v: A < S)
         def repeat(i: Int)(using ev: A => Unit, frame: Frame, flat: Flat.Weak[A]): Unit < S =
-            if i <= 0 then () else v.andThen(repeat(i - 1))
+            if i <= 0 then ()
+            else
+                // TODO Test failures in Scala Native without `return`
+                return v.andThen(repeat(i - 1))
```

Comment:
This was the most confusing issue I had. It seems Scala Native ends up returning `()` instead of the `Unit < S` computation without the `return` clause. I wonder if this issue could appear in other places.

Reply:
Is this just a type inference issue? Very strange.

Reply:
I'm not sure since I don't know how to inspect the code the compiler is generating. In this case, the `then` branch returns a `()` but the `else` returns a computation so the `()` should be lifted to `Unit < Any` via the `<.lift` implicit conversion. I've tried to reproduce it insolation but didn't have success. It could be a bad interaction between the multiple features Kyo uses: opaque types, extension methods, implicit conversions, inlining, etc. I don't think we should block on it since the tests are passing but we should have a ticket to follow up.